### PR TITLE
Ins 20862| Support kafka connect via api v2 tf

### PR DESCRIPTION
**Pull request by debaditya-netapp**
This is to update docs to support creating kafka connect clusters via provider-v2

### CODE REVIEW

File: docs/data-sources/kafka_cluster_v2_instance.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ The following terms are used to describe attributes in the schema of this data s
 <br>Status of the cluster.<br><br>
 *___kafka_version___*<br>
 <ins>Type</ins>: string, read-only<br>
-<ins>Constraints</ins>: pattern: `[0-9]+\.[0-9]+\.[0-9]+`<br><br>Version of Kafka to run on the cluster. Available versions: <ul> <li>`2.7.1`</li> <li>`2.8.1`</li> <li>`3.0.0`</li> <li>`3.1.1`</li> </ul><br><br>
+<ins>Constraints</ins>: pattern: `[0-9]+\.[0-9]+\.[0-9]+`<br><br>Version of Kafka to run on the cluster. Available versions: <ul> <li>`3.0.2`</li> <li>`3.1.2`</li> <li>`2.8.2`</li> <li>`2.7.1`</li> </ul><br><br>
```
Comment (YuWang24) [inline docs/data-sources/kafka_cluster_v2_instance.md:42]: Why the orders look so weird in the new file

File: docs/resources/kafka_connect_cluster_v2.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+<br>Creates the cluster with private network only, see [Private Network Clusters](https://www.instaclustr.com/support/documentation/useful-information/private-network-clusters/).<br><br>
+*___target_cluster___*<br>
+<ins>Type</ins>: nested block, required, immutable, see [target_cluster](#nested--target_cluster) for nested schema<br>
+<br>Details to connect to a target Kafka Cluster cluster.<br><br>
+### Input attributes - Optional
+*___custom_connectors___*<br>
+<ins>Type</ins>: nested block, optional, immutable, see [custom_connectors](#nested--custom_connectors) for nested schema<br>
+<br>Defines the location for custom connector storage and access info.<br><br>
+*___two_factor_delete___*<br>
+<ins>Type</ins>: nested block, optional, updatable, see [two_factor_delete](#nested--two_factor_delete) for nested schema<br>
```
Comment (YuWang24) [inline docs/resources/kafka_connect_cluster_v2.md:70]: Do we have an updatable feature of this in the app repo for Kafka Connect?

File: docs/resources/kafka_connect_cluster_v2.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+<ins>Type</ins>: string, read-only<br>
+<br>ID of the Cluster Data Centre.<br><br>
+*___nodes___*<br>
+<ins>Type</ins>: repeatable nested block, read-only, see [nodes](#nested--nodes) for nested schema<br>
+<br>
+<a id="nested--custom_connectors"></a>
+## Nested schema for `custom_connectors`
+Defines the location for custom connector storage and access info.<br>
+### Input attributes - Optional
+*___azure_settings___*<br>
```
Comment (YuWang24) [inline docs/resources/kafka_connect_cluster_v2.md:140]: Why do we have duplicated parameters like this?

File: docs/data-sources/kafka_connect_cluster_v2_instance.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+<ins>Type</ins>: nested block, read-only, see [azure_settings](#nested--azure_settings) for nested schema<br>
+<br>Defines the information to access custom connectors located in an azure storage container.<br><br>
+*___gcp_settings___*<br>
+<ins>Type</ins>: nested block, read-only, see [gcp_settings](#nested--gcp_settings) for nested schema<br>
+<br>Defines the information to access custom connectors located in a gcp storage container.<br><br>
+*___aws_settings___*<br>
+<ins>Type</ins>: nested block, read-only, see [aws_settings](#nested--aws_settings) for nested schema<br>
+<br>Defines the information to access custom connectors located in a S3 bucket.<br><br>
+<a id="nested--azure_settings"></a>
+## Nested schema for `azure_settings`
```
Comment (YuWang24) [inline docs/data-sources/kafka_connect_cluster_v2_instance.md:121]: Names are also duplicated here.

File: docs/data-sources/kafka_cluster_v2_instance.md (hunk 1 above)
Comment (debaditya-netapp) [reply docs/data-sources/kafka_cluster_v2_instance.md:42]: Thats the autogenerated doc by the tool. Its in a descending order.

File: docs/resources/kafka_connect_cluster_v2.md (hunk 2 above)
Comment (debaditya-netapp) [reply docs/resources/kafka_connect_cluster_v2.md:70]: Yes, Number of nodes, which comes under dataCentre. This is the same as the APIV2 signature

File: docs/resources/kafka_connect_cluster_v2.md (hunk 3 above)
Comment (debaditya-netapp) [reply docs/resources/kafka_connect_cluster_v2.md:140]: Not really, `awsSettings` here comes under `customConnectors`. The other one is under `dataCentre`, So the parent object here defines the context. The PD has the complete payload.

File: docs/data-sources/kafka_connect_cluster_v2_instance.md (hunk 4 above)
Comment (debaditya-netapp) [reply docs/data-sources/kafka_connect_cluster_v2_instance.md:121]: same as above

**Review by arjunrajnetapp — APPROVED** [review]

File: docs/resources/kafka_connect_cluster_v2.md (hunk 3 above)
Comment (debaditya-netapp) [reply docs/resources/kafka_connect_cluster_v2.md]: Hey Yu, I have updated the names of these.

**Review by sujayinstaclustr — APPROVED** [review]

File: docs/resources/kafka_connect_cluster_v2.md (hunk 2 above)
Comment (YuWang24) [reply docs/resources/kafka_connect_cluster_v2.md:69]: I mean this one ___two_factor_delete___, which is defined as updatable.

**Review by YuWang24 — APPROVED** [review]

**Pull request merged**